openai/openai.go

```diff
@@ -70,65 +104,40 @@ func (oac *OpenAiClient) GetCompletion(prompt string, debug bool) (*string, erro
 		fmt.Print(logMsg)
 	}
 
-	resp, err := oac.client.CreateChatCompletion(
-		context.Background(),
-		openai.ChatCompletionRequest{
-			Model: openai.GPT3Dot5Turbo,
-			Messages: []openai.ChatCompletionMessage{
-				{
-					Role:    openai.ChatMessageRoleSystem,
-					Content: systemMessage,
-				},
-				{
-					Role:    openai.ChatMessageRoleUser,
-					Content: prompt,
-				},
-			},
+	conversation := []openai.ChatCompletionMessage{
+		{
+			Role:    openai.ChatMessageRoleSystem,
+			Content: systemMessage,
 		},
-	)
+		{
+			Role:    openai.ChatMessageRoleUser,
+			Content: prompt,
+		},
+	}
 
+	firstDraft, err := oac.RequestCompletion(conversation)
 	if err != nil {
 		return nil, err
 	}
 
-	fmt.Printf(
-		"PromptTokens:\t\t%d\nCompletionTokens:\t%d\nTotalTokens:\t\t%d\n",
-		resp.Usage.PromptTokens,
-		resp.Usage.CompletionTokens,
-		resp.Usage.TotalTokens,
-	)
-	firstDraft := &resp.Choices[0].Message.Content
-
-	resp2, err2 := oac.client.CreateChatCompletion(
-		context.Background(),
-		openai.ChatCompletionRequest{
-			Model: openai.GPT3Dot5Turbo,
-			Messages: []openai.ChatCompletionMessage{
-				{
-					Role:    openai.ChatMessageRoleSystem,
-					Content: systemMessage,
-				},
-				{
-					Role:    openai.ChatMessageRoleUser,
-					Content: prompt,
-				},
-				{
-					Role:    openai.ChatMessageRoleAssistant,
-					Content: *firstDraft,
-				},
-				{
-					Role:    openai.ChatMessageRoleUser,
-					Content: followUpMessage,
-				},
-			},
+	conversation = append(conversation, []openai.ChatCompletionMessage{
+		{
+			Role:    openai.ChatMessageRoleAssistant,
+			Content: firstDraft,
 		},
-	)
-	if err2 != nil {
-		return nil, err2
+		{
+			Role:    openai.ChatMessageRoleUser,
+			Content: followUpMessage,
+		},
+	}...)
+
+	secondDraft, err := oac.RequestCompletion(conversation)
+	if err != nil {
+		return nil, err
 	}
-	secondResponse := &resp2.Choices[0].Message.Content
-	if strings.Contains(*secondResponse, "No changes") {
-		return firstDraft, nil
+
+	if strings.Contains(secondDraft, "No changes") {
+		return &firstDraft, nil
 	}
-	return &resp2.Choices[0].Message.Content, nil
```

Comment:
The GetCompletion function could be refactored to separate its responsibilities into smaller methods.